Get FITS-cubes tutorial out of quarantine and introduce reduced data cube

kakirastern opened the pull request:
Fixes #345 & fixes #289, as well as fixes #332

Goals: 
1. Update FITS-cubes tutorial in preparation of introduction of reduced data cube to be used with it. 
2. Upload reduced data cube file. 
3. Bring FITS-cubes out of quarantine.

kakirastern commented:
Things for repo maintainer(s) to do:
1. Upload the data file "reduced_TAN_C14.fits" to `http://data.astropy.org/tutorials/FITS-cubes/`
2. Don't delete the commits with the datafile. But instead, notify me so I can delete the redundant copy of "reduced_TAN_C14.fits" from my PR with an additional commit
3. I will update the tutorial path to `http://data.astropy.org/tutorials/FITS-cubes/` accordingly after the previous steps are completed
4. Then the PR is ready to merge

kakirastern commented:
For some reason Circle CI doesn't like the syntax used for quicklook() in Spectral-Cube. But Travis CI is passing as expected.

mwcraig commented on tutorials/notebooks/FITS-cubes/FITS-cubes.ipynb:
```diff
@@ -360,18 +362,20 @@
     "\n",
     "# Initiate a FITSFigure to set up axes\n",
-    "F = aplpy.FITSFigure(moment_1.hdu, figure=fig)\n",
+    "#F = aplpy.FITSFigure(moment_1.hdu, figure=fig)  # Please uncomment to initiate\n",
```
I think you need to uncomment the line above. The error at https://circleci.com/gh/astropy/astropy-tutorials/472?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link happens on line 371 below because `F` was never created because this line was commented out.

kakirastern replied:
Yeah, for some reason Circle CI doesn't like some of the `aplpy` syntax... I had to fix the error in a dumb way by commenting out all invocations of `F`... At least the test is passing now. Do you think there is something we can do with Circle CI to prevent such occurrences, i.e. the hyper-sensitivity to some non-standard package syntax?

eblur commented:
I checked out how the tutorial will be rendered on the website and I noticed a few minor issues. Can you please fix these, before I merge the PR:

* The bulleted list of conda install lines renders in a single line on the website. Could you please put them in a code block instead of a bulleted list? Like this:

```
conda install ...
conda install ...
```

* In the first blue box, the one-line method for reading in the spectral cube is commented out. Please remove the # symbol

* Why are the quick look plotting commands commented out? (Box 5 and 6) Please uncomment them. If they don't work with the reduced data-cube, then we need to modify this portion of the tutorial.

kakirastern commented:
Circle CI won't accept the code snippit
```
conda install -c astropy astroquery
conda install -c astropy spectral-cube
conda install -c astropy reproject
```
So will try 
```
!conda install -c astropy astroquery
!conda install -c astropy spectral-cube
!conda install -c astropy reproject
```
or something else instead.

kakirastern commented:
I have changed the path of the reduced data cube to `https://astropy.stsci.edu/data/tutorials/FITS-cubes/` in preparation of the anticipated merge... so now the error encountered on Circle CI is `HTTPError: HTTP Error 404: Not Found`, which will be fixed if the cube has indeed been moved to its designated new location. 

If I remember correctly, the quicklook lines were commented out because they were not passing Circle CI, due to its non-standard syntax.

eteq commented:
I merged astropy/astropy-data#64 so if you give it a few min (maybe up to an hour) it should start appearing in the correct location (if not ping me again as there's a manual step that is *sometimes* necessary).

One clarification on the above discussion: it is not sufficient to add a commit at the end of this PR that deletes the file, because it still is in the git history.  That would mean there's a pretty large data file in the git history which we are forever stuck with.  So removing that file requires a rebase, not just a "delete commit" at the end.  Let me know if help is needed with that process!

eteq commented:
Relatedly: it looks like there are a couple of  executed cells in the notebook here. So those will need to be cleared.  Normally I'd say that also requires a clean-up like above re: the data file, but the cells look to be just an exception traceback, so that's not really going to bloat the repo like e.g. a plot (which can be several MB in the notebook).

kakirastern commented:
No problem, I have rebased and dropped the commit originally adding the reduced data cube to this repo. Also, I have updated the FITS-cubes tutorials file to get rid of the unwanted traceback there due to an error. Am waiting for the reduced data cube to show up at the desired location. Thanks @eteq for the tips and clarification!

kakirastern commented:
Hi @eteq, is the moved reduced data cube supposed to show up here at [https://astropy.stsci.edu/data/tutorials/FITS-cubes/](https://astropy.stsci.edu/data/tutorials/FITS-cubes/)? I just checked and it is not there yet...

pllim commented:
Errr... Lemme check with @jhunkeler on the STScI server sync-ing.

kakirastern commented:
Thanks @pllim, that would be great.

pllim commented:
@kakirastern , the reduced cube is there now. Please let us know if you can't access it.

eteq commented:
The error is mysterious to me: it appears to be timing out while downloading `http://data.astropy.org/tutorials/synthetic-images/synchrotron_u_lobe_0700_150MHz_sm.fits`, but when I do that download locally all seems to be happy.  Can someone else (ideally not in the mid-atlantic of the US...) try the download so we can see if this might be some subtle locale problem?

mwcraig commented:
I was able to download it; it wasn't speedy, though, maybe 15-30 seconds?

pllim commented:
> it wasn't speedy

Physical server location and the resultant latency is out of our control unfortunately, unless we change the underlying infrastructure of how the data is hosted. 😬

eblur commented:
Notebook is breaking on the line:
```
spectrum = SDSS.get_spectra(plate=1349, fiberID=216, mjd=52797)[0]

...

URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:852)>
URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:852)>


Makefile:43: recipe for target 'html' failed
make: *** [html] Error 2
Exited with code 2
```

@kakirastern was able to duplicate this error on a local machine. We have opened [Issue #1436](https://github.com/astropy/astroquery/issues/1436) with astroquery.

kakirastern commented:
Reseting the timeout time to 60 seconds seems to have resolved the timeout issue. However, discovered a new error arising from some issue with `astroquery`, which led to an SSL error. Issue opened at astroquery to follow up on the matter: [https://github.com/astropy/astroquery/issues/1436](https://github.com/astropy/astroquery/issues/1436).
